cdap-common/src/main/java/co/cask/cdap/common/IncapableException.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common;
+
+import co.cask.cdap.api.annotation.Requirements;
+import co.cask.cdap.api.plugin.Plugin;
+import co.cask.cdap.runtime.spi.provisioner.Provisioner;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * An Exception which is thrown when a {@link Provisioner} is incapable of running a program because the program
+ * contains one or more {@link Plugin} whose {@link Requirements} are not meet by the provisioner.
+ */
+public class IncapableException extends ConflictException {
+  private final String programName;
+  private final String provisioner;
+  private final Map<String, Set<String>> unfulfilledRequirements;
+
+  /**
+   * Constructor
+   *
+   * @param programName the name of the program which is incapable to run
+   * @param provisioner the name of provisioner on which the program is incapable to run
+   * @param unfulfiledRequirements a {@link Map} of plugin name to {@link Set} of requirements for that plugin which the
+   * provisioner is unable to fulfill.
+   */
+  public IncapableException(String programName, String provisioner, Map<String, Set<String>> unfulfiledRequirements) {
+    super(String.format("Program '%s' cannot be run on provisioner '%s' The provisioner '%s' is incapable of " +
+                          "following plugin requirements'%s'", programName, provisioner, programName,
```

Comment:
Done. 
I think it is important to list down all the plugin names rather than showing only some so that the user knows all the plugins which are specifying the requirement. I did the aggregation by requirement. Sample output: 

```
2018-08-29 15:37:56,215 - ERROR [program.status:c.c.c.i.p.ProvisioningService@344] - 'DataPipelineWorkflow' cannot be run using profile 'native' because the profile does not meet all plugin requirements. Following requirements were not meet by the listed plugins: '{unicorn=[PluginId{name='IncapableSource', type='batchsource'}, PluginId{name='IncapableSink', type='batchsink'}]}'
```